exasol_script_languages_developer_sandbox/lib/aws_access.py

```diff
@@ -112,13 +117,145 @@ def delete_stack(self, stack_name: str) -> None:
+        """
+        logging.debug(f"Running create_image_from_ec2_instance for aws profile {self.aws_profile_for_logging}")
         cloud_client = self._get_aws_client("ec2")
-        return cloud_client.describe_instances(InstanceIds=[instance_id])["Reservations"][0]["Instances"][0]
+        tags = [{"ResourceType": "image", "Tags": create_default_asset_tag(tag_value)},
+                {"ResourceType": "snapshot", "Tags": create_default_asset_tag(tag_value)}]
+
+        result = cloud_client.create_image(Name=name, InstanceId=instance_id, Description=description,
+                                           NoReboot=False, TagSpecifications=tags)
+        return result["ImageId"]
+
+    def export_ami_image_to_vm(self, image_id: str, tag_value: str,
+                               description: str, role_name: str, disk_format: VmDiskImageFormat,
+                               s3_bucket: str, s3_prefix: str) -> None:
+        """
+        Creates an AMI image from an EC-2 instance
+        """
+        logging.debug(f"Running create_image_from_ec2_instance for aws profile {self.aws_profile_for_logging}")
+        cloud_client = self._get_aws_client("ec2")
+        tags = [{"ResourceType": "export-image-task", "Tags": create_default_asset_tag(tag_value)}]
+        result = cloud_client.export_image(ImageId=image_id, Description=description,
+                                           RoleName=role_name, DiskImageFormat=disk_format.value,
+                                           S3ExportLocation={"S3Bucket": s3_bucket, "S3Prefix": s3_prefix},
+                                           TagSpecifications=tags)
+
+        export_image_task_id, status, status_message = \
+            result["ExportImageTaskId"], result["Status"], result["StatusMessage"]
+        logging.info(f"Started export of vm image to {s3_bucket}/{s3_prefix}. Status message is {status_message}.")
+        last_progress = None
+        last_status = status
+        while status == "active":
+            time.sleep(10)
+            result = cloud_client.describe_export_image_tasks(ExportImageTaskIds=[export_image_task_id])
+            assert "NextToken" not in result #We expect only one result
+            export_image_tasks = result["ExportImageTasks"]
+            if len(export_image_tasks) != 1:
+                raise RuntimeError(f"Unexpected number of export image tasks: {export_image_tasks}")
+            export_image_task = export_image_tasks[0]
+            status = export_image_task["Status"]
+            status_message = get_value_safe("StatusMessage", export_image_task)
+            progress = get_value_safe("Progress", export_image_task)
+
+            if progress != last_progress or status != last_status:
+                logging.info(f"still running export of vm image to {s3_bucket}/{s3_prefix}. "
+                             f"Status message is {status_message}. Progess is '{progress}'")
+            last_progress = progress
+            last_status = status
```

Comment:
can we refactor this in multiple methods, may brain doesn't like more than 10 lines of code in a loop